llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -538,6 +542,12 @@ DemandedFields getDemanded(const MachineInstr &MI, const RISCVSubtarget *ST) {
     Res.MaskPolicy = false;
   }
 
+  if (isVExtractInstr(MI)) {
+    assert(!RISCVII::hasVLOp(TSFlags));
+    // TODO: LMUL can be any larger value (without cost)
```

Comment:
This would be an interesting optimisation. I could imagine us having more demanded LMUL types e.g.  `Res.LMUL = Res.LMULGreaterThanOrEqualToMX`

Reply:
In a sense, we sort of already have this for vmv.s.x and vmv.x.s, it's just that the MX value is the smallest legal LMUL, and thus GTE becomes any LMUL.